### ISSUE TITLE
Duplication Between Ontologies

### ISSUE THREAD

**Issue by VladimirAlexiev** [Sveino/Inst4CIM-KG#5]
Common terms are duplicated many times between ontologies.
See detailed analysis in rdfs-improvement/README:
- [Duplication Between Ontologies](#duplication-between-ontologies)
  - [Duplicated Definitions](#duplicated-definitions)
  - [Duplicated Terms](#duplicated-terms)
  - [Duplication Summary](#duplication-summary)

And the files mentioned below.
So these are just the counts:
```
wc -l *.txt
   882 duplicated-definitions.txt
   875 duplicated-terms.txt
  7268 terms-uniq.txt
```
The problem is pervasive: 12% of terms are duplicated (875 out of 7268).
The most "popular" terms are duplicated 28 times:
```
sort -rn duplicated-terms.txt |head -10
     28 cim:String
     28 cim:Date
     24 cim:IdentifiedObject.mRID
     24 cim:IdentifiedObject
     23 cim:Float
     22 cim:IdentifiedObject.name
     21 cim:UnitSymbol
     21 cim:UnitMultiplier
     21 cim:DateTime
     21 cim:Boolean
```
It's not only about primitives and other meta-terms. 
Electrical terms are also duplicated, eg:
```
     15 cim:ActivePower
     15 cim:ActivePower.multiplier
     15 cim:ActivePower.unit
     15 cim:ActivePower.value
```

What's the problem:
- [ ] There are discrepancies between multiple definitions. 
  - They are due to CGMES using `RDFSEd2Beta` style but NC using `RDFS2020` style and will be fixed by https://github.com/Sveino/Inst4CIM-KG/issues/41
- [ ] If you put ontologies in separate named graphs, there will be actual duplicated definitions of classes and properties, causing a lot more expensive reasoning
  - Is this a valid use case?
- [ ] Pedagogical: if you start reading profiles (eg first EQ then GL), you'll come across many things you've already read.

**Comment by VladimirAlexiev** [comment]
Eliminating duplication would require proper modularization, i.e. the creation of more ontology files: CIM datatypes, CIM core, CGMES core, etc.
- I think cims `Packages` already have this separation
- But it will require rethinking and changes to `CimSyntaxGen`: can't be done in SPARQL
- So I tag `longerm` and assign to @Sveino for consideration
- See `What's the problem:` in the description: maybe this is not a problem at all?

**Comment by VladimirAlexiev** [comment]
Linking is done by `owl:imports`.
- Eg EQ would import CIMS (datatypes: if needed), CORE (eg IdentifiedObject), etc.
- I would guess that such modularization is present in the UML model

But if the definitions are identical and the ontologies won't be loaded in named graphs, then the duplication does no harm.

**Comment by VladimirAlexiev** [comment]
From @Sveino's presentation `DX-PROF Balance vs Unbalance.pptx`:
![image](https://github.com/user-attachments/assets/23936279-694d-4ea3-a32c-9017943e0f50)

We discussed the idea that instead of 20 AP ontologies that define terms multiple times, we can have 40 ontologies that define each term once, `Core, Wires...` etc reused (imported) by `EQ, EQBD` etc. 
This should happen in CIM18 using CimContextor (for vocabulary profiling).

**Comment by VladimirAlexiev** [comment]
BTW I notice a little redundancy here (eg from 61970-600-2_Topology-AP-Voc-RDFS2020_v3-0-0.rdf):
```xml
    <rdfs:subClassOf>
      <rdfs:Class rdf:about="http://iec.ch/TC57/CIM100#ACDCTerminal"/>
    </rdfs:subClassOf>
```
This not only refers to `ACDCTerminal`, but also specifies its RDF type (which is repeated in the full description of that class).
Because every referenced class is redundantly defined in each ontology and formatted turtle eliminates duplicate triples, this smaller redundancy is not seen in turtle.

Other references don't have such redundancy, eg:
```xml
   <rdfs:range rdf:resource="http://iec.ch/TC57/CIM100#ConnectivityNode"/>
```

**Comment by VladimirAlexiev** [comment]
We agreed 2 weeks ago that "ontology modularization" is needed:
- only packages (eg Wires) should define terms
- profiles should only import packages (eg Equipment imports Wires) but not define terms

@Sveino @griddigit-ci right?

**Comment by Sveino** [comment]
Yes, this is the way I would like to have it for `61970-501:ED2`. But I am not sure if we need to have this done before we can finalize the work.
